corehq/apps/users/static/users/js/mobile_workers.js

```diff
@@ -0,0 +1,121 @@
+hqDefine("users/js/mobile_workers", function () {
+    var mobileWorker = function (options) {
+        var self = ko.mapping.fromJS(options);
+
+        self.mark_activated = ko.observable(false);
+        self.mark_deactivated = ko.observable(false);
+        self.action_error = ko.observable('');
+
+        self.showActivate = ko.computed(function () {
+            return self.action() === 'activate';
+        });
+
+        self.showDeactivate = ko.computed(function () {
+            return self.action() === 'deactivate';
+        });
+
+        self.modifyStatus = function (isActive, button) {
+            var urlName = isActive ? 'activate_commcare_user' : 'deactivate_commcare_user',
+                modalId = (isActive ? 'activate_' : 'deactivate_') + self.user_id();
+            $(button).addSpinnerToButton();
+            $.ajax({
+                method: 'POST',
+                url: hqImport("hqwebapp/js/initial_page_data").reverse(urlName, self.user_id()),
+                data: {
+                    is_active: isActive,
+                },
+                success: function (data) {
+                    $('#' + modalId).modal('hide');
+                    if (data.success) {
+                        self.mark_activated(isActive);
+                        self.mark_deactivated(!isActive);
+                        self.action_error('');
+                    } else {
+                        self.action_error(data.error);
+                    }
+                },
+                error: function () {
+                    $('#' + modalId).modal('hide');
+                    self.action_error(gettext("Issue communicating with server. Try again."));
+                },
+            });
+        };
+        return self;
+    };
+
+    var mobileWorkersList = function () {
+        var self = {};
+        self.users = ko.observableArray([]);
+
+        self.query = ko.observable('');
+        self.inactiveOnly = ko.observable(false);
+
+        self.itemsPerPage = ko.observable(5);
+        self.totalItems = ko.observable();
+
+        // Visibility of spinner, messages, and user table
+        self.hasError = ko.observable(false);
+        self.notLoaded = ko.observable(true);
+        self.projectHasUsers = ko.observable(true);
+
+        self.showProjectHasNoUsers = ko.computed(function () {
+            return !self.notLoaded() && !self.hasError() && !self.projectHasUsers();
+        });
+
+        self.showNoUsers = ko.computed(function () {
+            return !self.notLoaded() && !self.hasError() && !self.totalItems() && !self.showProjectHasNoUsers();
+        });
+
+        self.showSpinner = ko.computed(function () {
+            return self.notLoaded() && !self.hasError();
+        });
+
+        self.showTable = ko.computed(function () {
+            return !self.notLoaded() && !self.hasError() && !self.showNoUsers() && !self.showProjectHasNoUsers();
+        });
+
+        self.toggleInactiveOnly = function (inactiveOnly) {
```

Comment:
this function doesn't actually toggle the value, but sets it to the value of the argument. I'd call it `showInactiveOnly` or something similar. 
Alternatively, in the template, you could directly set the property of `inactiveOnly`, and set up a listener for a change to that value which calls `goToPage(1)`, i.e.: 
```js
self.inactiveOnly.subscribe(function () { self.goToPage(1); });
```

Reply:
a00971a